# Milestone 2: Insight Of Data And Feasibility Check Of Our Idea

<hr style="clear:both">

This notebook traces our journey through the feasibility check of our idea : Checking trends in music composition for the movie industry.

DONT FORGET TO DELETE:
**Guidline for us (to delete when submit the milestone):**
This notebook will present the journey of the group through the data analysis. We show that we :
- can handle the data in its size.
- understand what’s in the data giving the formats, distributions, missing values, correlations, and so on.
- considered ways to enrich, filter, transform the data according to our needs.
- have a reasonable plan and ideas for methods we’re going to use, giving their essential mathematical details in the notebook.
- have a plan for analysis and its communication is reasonable and sound, potentially discussing alternatives to our choices that we considered but dropped.

**Project Mentor:** [Aoxiang Fan](https://people.epfl.ch/aoxiang.fan) ([Email](mailto:aoxiang.fan@epfl.ch))
**Authors:** [Luca Carroz](https://people.epfl.ch/emilie.carroz), [David Schroeter](https://people.epfl.ch/david.schroeter), [Xavier Ogay](https://people.epfl.ch/xavier.ogay), [Joris Monnet](https://people.epfl.ch/joris.monnet), [Paulo Ribeiro de Carvalho](https://people.epfl.ch/paulo.ribeirodecarvalho)

<hr style="clear:both">

## Story-line

**Title : Helping Stanislas to achieve his dream !**

A 20-year-old aspiring musician, Stanislas, fueled by a passion for the film industry, embarks on a quest to launch his career. His ultimate dream? To hear one of his productions featured in a Hollywood film and become one of the planet's top composers. To increase his chances, he turns to a team of Data Scientists known as LSD.

The LSD team's mission is to identify trends shared among successful music composers and compositions, ultimately optimizing choices for our young musician. The storyline focuses on American movies and Successful composers (successful meaning those who have produced soundtracks for high-grossing films, music appearing in more than ?? films, etc.)


## Needed data

To help Stanislas with his dream, we will need some specific information about each analysed movie. To support our final answer concerning potential trends in composer for movie industry, we will analyse few questions :
1) Which are the most frequent music genre appearing in movies ?
2) What is the average composer's age at their first movie appearance ?
3) Where do composers come from ?
4) Does composer's gender matter ?
5) Does personal website is preferred ?
6) What is the expected **fame** to appear in a movie ? (Challenging : **fame** can be quantified with the amount of streams or CD selling for older compositors.)

List of needed information about movie's composer :
- Name
- Birthday
- Gender
- Homepage
- Place of birth
- First appearance in movie

## Import

In [1]:
# import all needed libraries
from helpers import *

# Load autoreload extension
%load_ext autoreload

# Set autoreload mode
%autoreload 2

## Load Data

Only `movie.metadata.tsv` is loaded, since others data sets are not relevant for our data analysis.

Note that no header are given in the raw data sets. We then looked at the documentation and set these headers with meaningful names. Please, find the documentation of data set [here](http://www.cs.cmu.edu/~ark/personas/).

In [2]:
# Please be sure your data is store in the same path
data_path = 'dataset/MovieSummaries/movie.metadata.tsv'

# Load the text file into Pandas DataFrame
movie = load_movies(data_path)

# Display the dataframe
display(movie)

wiki_movieID freebase_movieID  \
0            975900        /m/03vyhn   
1           3196793        /m/08yl5d   
2          28463795       /m/0crgdbh   
3           9363483       /m/0285_cd   
4            261236        /m/01mrr1   
...             ...              ...   
81736      35228177       /m/0j7hxnt   
81737      34980460       /m/0g4pl34   
81738       9971909       /m/02pygw1   
81739        913762        /m/03pcrp   
81740      12476867       /m/02w7zz8   

                                                    name release_date  \
0                                         Ghosts of Mars   2001-08-24   
1      Getting Away with Murder: The JonBenét Ramsey ...   2000-02-16   
2                                            Brun bitter         1988   
3                                       White Of The Eye         1987   
4                                      A Woman in Flames         1983   
...                                                  ...          ...   
81736                           Mermaids: The Body Found   2011-03-19   
81737                                            Knuckle   2011-01-21   
81738                                  Another Nice Mess   1972-09-22   
81739  The Super Dimension Fortress Macross II: Lover...   1992-05-21   
81740                                            Spliced         2002   

       box_office_revenue  runtime                           languages  \
0              14010832.0     98.0  {"/m/02h40lc": "English Language"}   
1                     NaN     95.0  {"/m/02h40lc": "English Language"}   
2                     NaN     83.0  {"/m/05f_3": "Norwegian Language"}   
3                     NaN    110.0  {"/m/02h40lc": "English Language"}   
4                     NaN    106.0   {"/m/04306rv": "German Language"}   
...                   ...      ...                                 ...   
81736                 NaN    120.0  {"/m/02h40lc": "English Language"}   
81737                 NaN     96.0  {"/m/02h40lc": "English Language"}   
81738                 NaN     66.0  {"/m/02h40lc": "English Language"}   
81739                 NaN    150.0   {"/m/03_9r": "Japanese Language"}   
81740                 NaN     86.0  {"/m/02h40lc": "English Language"}   

                                               countries  \
0              {"/m/09c7w0": "United States of America"}   
1              {"/m/09c7w0": "United States of America"}   
2                                 {"/m/05b4w": "Norway"}   
3                         {"/m/07ssc": "United Kingdom"}   
4                                {"/m/0345h": "Germany"}   
...                                                  ...   
81736          {"/m/09c7w0": "United States of America"}   
81737  {"/m/03rt9": "Ireland", "/m/07ssc": "United Ki...   
81738          {"/m/09c7w0": "United States of America"}   
81739                              {"/m/03_3d": "Japan"}   
81740                            {"/m/0d060g": "Canada"}   

                                                  genres  
0      {"/m/01jfsb": "Thriller", "/m/06n90": "Science...  
1      {"/m/02n4kr": "Mystery", "/m/03bxz7": "Biograp...  
2      {"/m/0lsxr": "Crime Fiction", "/m/07s9rl0": "D...  
3      {"/m/01jfsb": "Thriller", "/m/0glj9q": "Erotic...  
4                                {"/m/07s9rl0": "Drama"}  
...                                                  ...  
81736                            {"/m/07s9rl0": "Drama"}  
81737  {"/m/03bxz7": "Biographical film", "/m/07s9rl0...  
81738       {"/m/06nbt": "Satire", "/m/01z4y": "Comedy"}  
81739  {"/m/06n90": "Science Fiction", "/m/0gw5n2f": ...  
81740  {"/m/01jfsb": "Thriller", "/m/03npn": "Horror"...  

[81741 rows x 9 columns]

A quick visual insight can be done to get information gain. We see that some variables are textual (i.e. `movie_freebase_ID`, `actor_name`, ...), categorical (i.e. `actor_gender`) or numerical (i.e. `actor_height`, `actor_age_movie_release`, ...).

Let's have a more meaningful insight of our current data.

In [3]:
# TODO : Finish the insight function to give a better display of information about dataframe
insight(movie)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 81741 entries, 0 to 81740
Data columns (total 9 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   wiki_movieID        81741 non-null  int64  
 1   freebase_movieID    81741 non-null  object 
 2   name                81741 non-null  object 
 3   release_date        74839 non-null  object 
 4   box_office_revenue  8401 non-null   float64
 5   runtime             61291 non-null  float64
 6   languages           81741 non-null  object 
 7   countries           81741 non-null  object 
 8   genres              81741 non-null  object 
dtypes: float64(2), int64(1), object(6)
memory usage: 5.6+ MB


(None,
        wiki_movieID  box_office_revenue       runtime
 count  8.174100e+04        8.401000e+03  6.129100e+04
 mean   1.740784e+07        4.799363e+07  1.118192e+02
 std    1.098791e+07        1.121753e+08  4.360070e+03
 min    3.300000e+02        1.000000e+04  0.000000e+00
 25%    7.323695e+06        2.083193e+06  8.100000e+01
 50%    1.777899e+07        1.063969e+07  9.300000e+01
 75%    2.715573e+07        4.071696e+07  1.060000e+02
 max    3.750192e+07        2.782275e+09  1.079281e+06)

We see that only few movies (8,401) have information about their box office revenue. Since it is an important metric for us, we will only be able to analyze these movies.
One solution would be to complete the dataframe with another dataframe.

More importantly, we see that needed attributes are not given. We then need to retrieve them via external datasets. To do so, we use a free to use API ([TMDB](https://www.themoviedb.org/?language=fr)) to enrich our movies' information.

A specific script has been created to be run once and create our `clean_enrich_movie.csv` dataset. Go to `enrich_data.py` and its linked library `tmdb/tmdb.py` for more explanation on how we retrieved these information. Please note that a personal API key is needed to successfully run the script ([create key](https://developer.themoviedb.org/reference/intro/getting-started)). Make sure to create file `.env` with your API bearer token using the `.env_example` as template.

## Enrich Data Set

Let's enrich our dataset with an additional one downloaded [here (LINK NOT PROVIDED YET)](https://google.com).

Here we should also provide some statistical inisght of this new dataset to be sure that it is interesting for us to work with this one.

In [4]:
pass

## Joining Data Sets

Explain how we are going to join the datasets to have a meaningful big dataset to achieve our goal.

In [5]:
pass

## Results Expected

Let's try to explain what are the main plots we want to bring to our readers. Do we have in mind also to have a interactive plot, and so on ?

In [6]:
pass